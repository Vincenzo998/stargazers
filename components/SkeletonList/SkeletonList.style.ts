import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  card: {
    height: 80,
    backgroundColor: "#f5f5f5",
    marginHorizontal: "5%",
    marginBottom: "5%",
    borderRadius: 5,
  },
  animation: { width: "10%", height: "100%", backgroundColor: "#ffffff", opacity: 0.5 },
});

export default styles;
