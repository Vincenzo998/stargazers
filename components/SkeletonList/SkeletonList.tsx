import React, { useCallback, useEffect } from "react";
import { Animated } from "react-native";

// styles
import styles from "./SkeletonList.style";

const SkeletonList = () => {
  const skeletonAnimate = new Animated.Value(0);

  const translateX = skeletonAnimate.interpolate({
    inputRange: [0, 1],
    outputRange: [-10, 400],
  });

  const animate = useCallback(() => {
    skeletonAnimate.setValue(0);
    Animated.timing(skeletonAnimate, {
      toValue: 1,
      duration: 300,
      useNativeDriver: false,
    }).start(() => {
      setTimeout(() => animate(), 900);
    });
  }, []);

  useEffect(() => {
    animate();
  }, [animate]);

  return (
    <>
      <Animated.View style={{ ...styles.card, marginTop: "8%" }}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
      <Animated.View style={styles.card}>
        <Animated.View style={{ ...styles.animation, transform: [{ translateX }] }} />
      </Animated.View>
    </>
  );
};

export default SkeletonList;
