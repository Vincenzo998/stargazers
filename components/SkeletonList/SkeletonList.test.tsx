import React from "react";
import renderer from "react-test-renderer";

// screens
import SkeletonList from ".";

test("SkeletonList should render without crashing", () => {
  const SkeletonListComponent = renderer.create(<SkeletonList />).toJSON();
  expect(SkeletonListComponent).toMatchSnapshot();
});
