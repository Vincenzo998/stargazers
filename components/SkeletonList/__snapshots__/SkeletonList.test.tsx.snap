// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonList should render without crashing 1`] = `
[
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
        "marginTop": "8%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 5,
        "height": 80,
        "marginBottom": "5%",
        "marginHorizontal": "5%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0.5,
          "transform": [
            {
              "translateX": -10,
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>,
]
`;
